huaweicloud/services/dew/resource_huaweicloud_kms_key.go

```diff
@@ -369,29 +392,31 @@ func resourceKmsKeyV1Update(d *schema.ResourceData, meta interface{}) error {
 			KeyID:    keyID,
 			Interval: d.Get("rotation_interval").(int),
 		}
-		err := rotation.Update(kmsKeyV1Client, intervalOpts).ExtractErr()
+		err := rotation.Update(client, intervalOpts).ExtractErr()
 		if err != nil {
 			return fmt.Errorf("failed to change key rotation interval: %s", err)
 		}
 	}
 
-	return resourceKmsKeyV1Read(d, meta)
+	return nil
 }
 
-func resourceKmsKeyV1Delete(d *schema.ResourceData, meta interface{}) error {
+func resourceKmsKeyV1Delete(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	config := meta.(*config.Config)
-	kmsKeyV1Client, err := config.KmsKeyV1Client(GetRegion(d, config))
+	region := config.GetRegion(d)
+	kmsKeyV1Client, err := config.KmsKeyV1Client(region)
 	if err != nil {
-		return fmt.Errorf("error creating KMS key client: %s", err)
+		return diag.Errorf("error creating KMS key client: %s", err)
 	}
 
 	v, err := keys.Get(kmsKeyV1Client, d.Id()).ExtractKeyInfo()
 	if err != nil {
-		return CheckDeleted(d, err, "failed to retrieve key")
+		return common.CheckDeletedDiag(d, err, "failed to retrieve key")
 	}
 
 	deleteOpts := &keys.DeleteOpts{
-		KeyID: d.Id(),
+		KeyID:       d.Id(),
+		PendingDays: "7",
```

Comment:
it seems that we can use `d.Get("pending_days").(string)` directly.

Reply:
pending_days just been used by delete function, so remove the default value from schema.